File: app/src/main/java/com/idtk/mylink/MainActivity.java
package com.idtk.mylink;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toast.makeText(this, "toast", Toast.LENGTH_SHORT);

        Log.d("tag", "msg");
        ABean a =  new ABean();
        a.setName("111");

    }


}

File: app/src/main/java/com/idtk/mylink/MainActivity.kt
package com.idtk.mylink

import android.app.Application
import android.app.Activity
import android.os.Bundle
import android.Manifest.permission
import android.util.Log
import android.os.Looper

class MainActivity : Activity() {


    private lateinit var test:String
    private var test2:String ? = null
    private var b : AData.BData ?= null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        test = "msg"
        test2 = "msg2"

        val a = ABean()
        a.name = "111"
        a.list

        Log.d("tag11",test2!!)
        val ad = AData()
        b = AData.BData()
        val c = AData.BData.CData()
        b!!.c = c
        ad.b = b
    }

}
